x/accesscontrol/keeper/keeper.go

```diff
@@ -171,18 +171,80 @@ func (k Keeper) GetWasmDependencyAccessOps(ctx sdk.Context, contractAddress sdk.
 
 	selectedAccessOps, err := k.BuildSelectorOps(ctx, contractAddress, accessOps, senderBech, msgInfo, circularDepLookup)
 	if err != nil {
+		// TODO: should we reset wasm dependency mapping here?
 		return nil, err
 	}
-	return selectedAccessOps, nil
+
+	// imports base contract references
+	contractRefs := dependencyMapping.BaseContractReferences
+	// add the specific execute or query contract references based on message type + name
+	specificContractRefs := []*acltypes.WasmContractReferences{}
+	if msgInfo.MessageType == acltypes.WasmMessageSubtype_EXECUTE && dependencyMapping.ExecuteAccessOps != nil {
```

Comment:
this should be a len check, not a nil check right? because we would have a ezro length array instead of a nil pointer right?